modules/aws-eks-self-managed-node-groups/self-managed-launch-templates.tf

```diff
@@ -8,7 +8,7 @@ module "launch_template_self_managed_ng" {
       launch_template_os     = local.self_managed_node_group["launch_template_os"]
       launch_template_prefix = local.self_managed_node_group["node_group_name"]
       instance_type          = local.self_managed_node_group["instance_type"]
-      capacity_type          = local.self_managed_node_group["capacity_type"]
+      capacity_type          = local.self_managed_node_group["capacity_type"] == "spot" ? "" : local.self_managed_node_group["capacity_type"]
```

Comment:
why do we need to make this change - shouldn't it work as its written currently?

Reply:
when you're creating a spot ASG with mixed instance policies you need to leave it blank